File: replication/src/test/com/db4o/test/other/Db4oListTest.java
/* Copyright (C) 2004 - 2006 db4objects Inc. http://www.db4o.com */

package com.db4o.test.other;

import com.db4o.replication.db4o.Db4oReplicationProvider;
import com.db4o.types.Db4oList;


public class Db4oListTest extends ListTest {

	public void test() {
		if (!(a().provider() instanceof Db4oReplicationProvider))
			return;

		super.actualTest();
	}

	protected ListHolder createHolder() {
		ListHolder lh = new ListHolder("h1");
		Db4oList list = ((Db4oReplicationProvider) a().provider()).getObjectContainer().collections().newLinkedList();

		lh.setList(list);
		return lh;
	}

}

File: drs/drs.net/drs.tests/src/com/db4o/drs/test/Db4oListTest.cs
namespace com.db4o.drs.test
{
	public class Db4oListTest : com.db4o.drs.test.ListTest
	{
		public override void Test()
		{
			if (!(A().Provider() is com.db4o.drs.db4o.Db4oReplicationProvider))
			{
				return;
			}
			base.ActualTest();
		}

		protected override com.db4o.drs.test.ListHolder CreateHolder()
		{
			com.db4o.drs.test.ListHolder lh = new com.db4o.drs.test.ListHolder("h1");
			com.db4o.types.Db4oList list = ((com.db4o.drs.db4o.Db4oReplicationProvider)A().Provider
				()).GetObjectContainer().Collections().NewLinkedList();
			lh.SetList(list);
			return lh;
		}
	}
}
